### PR TITLE
Dev/fix data type union

### PR DESCRIPTION
**Pull request by jwittner**
Fixes a couple of the errant unions here: https://trac.osgeo.org/gdal/ticket/6879

### CODE REVIEW

File: gdal/gcore/gdal_misc.cpp [hunk 1]
```diff
@@ -79,6 +89,34 @@ static int GetMinBitsForPair(const bool pabSigned[], const int panBits[])
     return std::max(panBits[0], panBits[1]);
 }
 
+int GetDataTypeElementSizeBits( GDALDataType eDataType )
```
Comment (rouault) [inline gdal/gcore/gdal_misc.cpp]: Missing static qualifier.
Will fix "gdal_misc.cpp:92:5: error: no previous prototype for function 'GetDataTypeElementSizeBits' [-Werror,-Wmissing-prototypes]"

Comment (jwittner) [inline gdal/gcore/gdal_misc.cpp]: Thanks!

**Comment by jwittner** [comment]
@rouault Any idea what's up with the failed apt-gets?

**Comment by rouault** [comment]
> Any idea what's up with the failed apt-gets?

Unrelated to your changes. An issue with a ppa we use and that has a problem since a few days. I've emailed the maintainer. Will probably have to disable it if things don't go back soon

**Comment by rouault** [comment]
Applied in r38201

**Comment by jwittner** [comment]
Thanks!

**Comment by schwehr** [comment]
@jwittner GetMinBitsForPair really needs a comment explaining the algorithm.

**Pull request closed without merging**